core/src/main/java/com/graphhopper/storage/GraphEdgeIdFinder.java

```diff
@@ -165,23 +175,26 @@ public BlockArea(Graph g) {
         }
 
         public boolean hasCachedEdgeIds(int shapeIndex) {
-            return !edgesList.get(shapeIndex).isEmpty();
+            return edgesList.get(shapeIndex) != null;
         }
 
         public String toString(int shapeIndex) {
+            GHIntHashSet blockedEdges = edgesList.get(shapeIndex);
+            if (blockedEdges == null) {
+                return Collections.emptyList().toString();
+            }
+            
             List<Integer> returnList = new ArrayList<>();
-            for (IntCursor intCursor : edgesList.get(shapeIndex)) {
+            for (IntCursor intCursor : blockedEdges) {
```

Comment:
Are these changes necessary anymore? Because the edgesList contains now only HashSets that are != null and with size > 0 after your recent changes. (?)

Reply:
Yes. We either store `null` HashSets for large Shapes or non-empty ones for small shapes.

Reply:
Yes. We either store a `null` HashSet for large shapes or non-empty HashSets for smaller ones. We only drop empty HashSets for small shapes.

Reply:
> We either store a null HashSet for large shapes

Why not an empty set where we can avoid the special `null` handling?